File: psp/card-service/src/main/java/com/psp/cardservice/dto/BankInitPaymentRequest.java
package com.psp.cardservice.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BankInitPaymentRequest {
    private String merchantId;
    private String pspPaymentId;
    private Double amount;
    private String currency;
    private String stan;
    private Long pspTimestamp;
}

File: psp/card-service-csharp/CardService.Domain/DTOs/BankInitPaymentRequest.cs
using CardService.Domain.Enums;

namespace CardService.Domain.DTOs
{
    public record BankInitPaymentRequest(
        string MerchantId,
        Guid PspPaymentId,
        double Amount,
        Currency Currency,
        string Stan,
        DateTime PspTimestamp
    );
}
